lib/parallel.rb

```diff
@@ -267,20 +271,18 @@ def work_in_processes(items, options, &blk)
               item, index = items.next
               break unless index
 
-              output = with_instrumentation item, index, options do
-                worker.work(items.pack(item, index))
-              end
-
-              if ExceptionWrapper === output
-                exception = output.exception
+              begin
+                results[index] = with_instrumentation item, index, options do
+                  worker.work(items.pack(item, index))
+                end
+              rescue StandardError => e
```

Comment:
The exception (DeadWorker or otherwise) will still get re-raised in `handle_exception`. The `it "raises DeadWorker..."` spec covers that.